### PR TITLE
Feature/14 reservation api actions

### LINKED ISSUE #14: Reservation actions to api module
Add actions for:
- [x] fetching reservations
- [x] making a reservation
- [x] editing a reservation
- [x] cancelling a reservation

### PR DESCRIPTION
**Pull request by lraivio**
Built on top of #13 so that should be merged first.

This PR:
- Adds api actions for fetching, making, editing and cancelling reservations.
- Refactors defining api actionTypes.
- Fixes some wrong api action test describe texts.

The dataReducer could benefit from some refactoring, but that can probably be left to future so we stay in schedule.

Closes #14.

### CODE REVIEW

File: app/api/actions/reservations.js [hunk 1]
```diff
@@ -11,6 +14,21 @@ function fetchReservations(params = {}) {
...
+  return createApiAction({
+    endpoint: 'reservation',
+    method: 'POST',
+    type: 'RESERVATION',
+    body: parseReservationData(reservation),
+  });
+}
+
+function parseReservationData(reservation) {
+  const parsed = pickBy(reservation, value => value);
```
Comment (YoYuUm) [inline app/api/actions/reservations.js]: would `parsed` be the same than reservation but without keys which value is `null` or `undefined` ?

File: app/api/reducers/dataReducer.js [hunk 2]
```diff
@@ -55,6 +55,11 @@ function dataReducer(state = initialState, action) {
       return handleData(state, { resources });
     }
 
+    case actionTypes.RESERVATION_DELETE_SUCCESS: {
+      const reservation = Object.assign({}, action.payload, { state: 'cancelled' });
```
Comment (YoYuUm) [inline app/api/reducers/dataReducer.js]: - [ ] Doesn't the API return cancelled state? Can we receive later on cancelled reservations from the backend?

File: app/api/actions/reservations.js (hunk 1 above)
Comment (lraivio) [inline app/api/actions/reservations.js]: The API will throw an error if reservation is saved or edited with empty string values. Saving 0 values should be made possible so I'll change that at least. I'll have to test with null values but depending on the API that could be valid use case.

File: app/api/reducers/dataReducer.js (hunk 2 above)
Comment (lraivio) [inline app/api/reducers/dataReducer.js]: Cancelling is done with a DELETE request and the API doesn't return anything with it. In the backend reservations are soft deleted so their state is just turned into 'cancelled'.
The payload here is the original reservation and put there by us in the deleteReservation action. I guess the state change could be made in the action as well if that would be more clear?
When fetching reservations or resources, the API also returns cancelled (deleted) reservations.

File: app/api/actions/reservations.js (hunk 1 above)
Comment (lraivio) [inline app/api/actions/reservations.js]: I checked that the API doesn't work with null values either. But I changed the parse function to allow values that are 0.

**Comment by lraivio** [comment]
Changes made. I also removed body from DELETE reservation api call. That should not have been there.

**Pull request merged**